src/atomate2/qchem/sets/core.py

```diff
@@ -0,0 +1,132 @@
+"""Module defining core QChem input set generators"""
+
+from __future__ import annotations
+
+import logging
+from copy import deepcopy
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Optional, Tuple
+
+import numpy as np
+from pymatgen.core.structure import Molecule
+
+from atomate2.qchem.sets.base import QChemInputGenerator
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "SinglePointSetGenerator",
+    "OptSetGenerator",
+    "TransitionStateSetGenerator",
+    "ForceSetGenerator",
+    "FreqSetGenerator",
+    "PESScanSetGenerator",
+]
+
+@dataclass
+class SinglePointSetGenerator(QChemInputGenerator):
+    """Class to generate QChem Single Point input sets."""
+
+    def get_basis_set_updates(
+            self,
+            molecule: Molecule,
+            prev_basis: str = None,
+            prev_scf: str = None,
+            new_geom_opt: dict = None,
+            nbo_params: dict = None,
+            overwrite_inputs: dict = None, 
+        ) -> dict:
+            """
+            Get updates to the basis set for a single point calculation.
+            """
+            return {"basis_set": "def2-tzvppd", "scf_algorithm": "diis", "job_type": "sp"}
+
```

Comment:
Each of the `QChemInputGenerator` is implemented with a `get_basis_set_updates` method that returns a pre-populated dict. Wouldnt it be simpler to just do this directly, e.g.

```
class SinglePointSetGenerator(QChemInputGenerator):
    """Class to generate QChem Single Point input sets."""
    return QChemInputGenerator("basis_set"="def2-tzvppd",
                                                     "scf_algorithm"="diis",
                                                     "job_type": "sp")
'''

Reply:
In the simplest form, I had not even envisioned having any inheritance in the `InputSetGenerator`. E.g. the only function of the Generator would be to define specific sets of kwargs that get passed to `InputSet`

```
class OptSetGenerator(InputGenerator):

     def __init__(basis_set, scf_algorithm, dft_rung, etc.)
           ....

     def get_input_set(Molecule):

           return `QChemInputSet(molecule, basis_set, scf_algorithm, dft_rung).

```

There may be some technical advantages to have the intermediate class `QChemInputGenerator`, especially with regard to inheriting previous calc directories. But if we can possibly do without it, I'd advocate for the approach above. Keep things simple!

@utf I know VASP makes use of the dual layer model and I think that's what @rdguha1995 is emulating here. Do you have any additional thoughts on when/why that's advantageous?